app/models/user_challenge/evaluator.rb

```diff
@@ -0,0 +1,43 @@
+class UserChallenge
+  class Evaluator
+    attr_reader :ids_to_comments, :ids_to_places, :user_challenge_ids
+
+    def initialize(params:)
+      permitted_params = params.permit(user_challenges: [:id, :comments, :place])
+
+      @user_challenge_ids = permitted_params['user_challenges'].pluck(:id)
+      @ids_to_comments = permitted_params['user_challenges'].each_with_object({}) do |val, acc|
+        acc[val['id']] = val['comments']
+        acc
+      end
+      @ids_to_places = permitted_params['user_challenges'].each_with_object({}) do |val, acc|
+        acc[val['id']] = val['place']
+        acc
+      end
+    end
+
+    def save_comments
+      save_a_field(field_name: :comments, id_map: ids_to_comments)
+    end
+
+    def save_places
+      save_a_field(field_name: :place, id_map: ids_to_places)
+    end
+
+    private
+
+    def save_a_field(field_name:, id_map:)
```

Comment:
Yup!

`def method_name(arg_name:)`:
- In this case, `arg_name` is a required keyword argument. Meaning if I don't call the method like `method_name(arg_name: some_arg_here)`, it'll throw an `ArgumentError`.

`def method_name(arg_name: nil)`:
- In this case, `arg_name` is a keyword argument with a default value (of `nil`), so it is optional.

[Here's](https://robots.thoughtbot.com/ruby-2-keyword-arguments) a nice blog post on them.